disconnected_registry/docker-registry-sync.py

```diff
@@ -61,18 +70,39 @@
 latest_tag_list = []
 failed_images = []
 
-
-def generate_url_list(dictionary_key, list_to_populate):
+# Authenticate with docker v2 compatible registry, return access token response by auth server to the caller.
+def get_registry_access_token(user, tokenpass, auth_url):
+    auth_session = requests.Session()
+    b64_userpass = base64.b64encode("%s:%s" % (user, tokenpass))
+    auth_header = {'Authorization': 'Basic ' + b64_userpass}
```

Comment:
If pulled from `docker.json` the encoding step is not needed

Reply:
Strongly agree, no need to encode when auth in config.json already in base64 encoded.